python/foglamp/services/core/api/filters.py

```diff
@@ -0,0 +1,264 @@
+                allow_duplicates = request.query['allow_duplicates'].lower()
+                if allow_duplicates not in ['true', 'false']:
+                    raise ValueError("Only 'true' and 'false' are allowed for "
+                                     "allow_duplicates. {} given.".format(allow_duplicates))
+
+            if append_filter == 'true':
+                # 'value' holds the string version of a list: convert it first  
+                current_value = json.loads(category_info[config_item]['value'])
+                # Save current list (deepcopy)
+                new_list = copy.deepcopy(current_value['pipeline'])
+                # iterate inout filters list
+                for _filter in filter_list:
+                    # Check whether we need to add this filter
+                    if allow_duplicates == 'true' or (_filter not in current_value['pipeline']):
+                        # Add the new filter to new_list
+                        new_list.append(_filter)
+            else:
+                # Overwriting the list: use input list
+                new_list = filter_list
+
+            # Set the pipeline value with the 'new_list' of filters
+            await cf_mgr.set_category_item_value_entry(service_name,
+                                                       config_item,
+                                                       {'pipeline': new_list})
+        else:
+            # Create new item 'config_item'
+            new_item = dict({config_item: {
+                'description': 'Filter pipeline',
+                'type': 'JSON',
+                'default': {}
+            }
+            })
+
+            # Add the "pipeline" array as a string
+            new_item[config_item]['default'] = json.dumps({'pipeline': filter_list})
+
+            # Update the filter category entry
+            await cf_mgr.create_category(category_name=service_name,
+                                         category_value=new_item,
+                                         keep_original_items=True)
+
+        # Fetch up-to-date category items
+        result = await cf_mgr.get_category_item(service_name, config_item)
+        if result is None:
+            # Error config_item doesn't exist
+            message = "No detail found for the category_name: {} " \
+                      "and config_item: {}".format(service_name, config_item)
+            return web.HTTPNotFound(reason=message)
+
+        ###########################################################
+        # Add filters as child categories of parent category name #
+        ###########################################################
+        await cf_mgr.create_child_category(service_name, filter_list)
+
+    except ValueError as ex:
+        _LOGGER.exception("Add filters pipeline, caught exception: " + str(ex))
+        raise web.HTTPNotFound(reason=str(ex))
+ 
+    # Return the filters pipeline 
```

Comment:
only catched ValueError exception, what if other exception occurred? Please fix it for both def endpoints
```
except Exception as ex:
         _LOGGER.exception("Add filters pipeline, caught exception: " + str(ex))
         raise web.HTTPInternalServer(reason=str(ex)
```